Module not found: Error: Cannot resolve module 'module' in [...]source-map-support @ source-map-support.js 464:17-34

I am using `source-map-support` `0.4.5` in a module of mine (let's call it `moduleA`).

I then publish `moduleA` to npm and all's well till that point.

Then I try to use that `moduleA` in another project of mine (let's call it `moduleUser`). After I do `npm install` in `moduleUser` I try to do an `npm run build` which is configured in `package.json` as:

```
$ grep \"build\"\: package.json 
    "build": "webpack --progress --colors",
```

&hellip; and I then encounter (at the end of the build process) the following message which I can't decipher:

ERROR in ../~/moduleA/~/source-map-support/source-map-support.js
Module not found: Error: Cannot resolve module 'module' in /home/carsten/moduleUser/node_modules/moduleA/node_modules/source-map-support
 @ ../~/moduleA/~/source-map-support/source-map-support.js 464:17-34

Looking in the identified location in file `source-map-support.js` the line is:

```
var Module = require('module');
```

I am using Angular 5, Angular cli 1.5.2 and everything was fine until I added and used 
`
import * as ts from "typescript";
`

My error starts with
```
WARNING in ./node_modules/source-map-support/source-map-support.js
Module not found: Error: Can't resolve 'module' in '/usr/local/app/node_modules/source-map-support'
resolve 'module' in '/usr/local/app/node_modules/source-map-support'
  Parsed request is a module
  using description file: /usr/local/app/node_modules/source-map-support/package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: /usr/local/app/node_modules/source-map-support/package.json (relative path: .)
    resolve as module
      /usr/local/app/node_modules/source-map-support/node_modules doesn't exist or is not a directory
      /usr/local/app/node_modules/node_modules doesn't exist or is not a directory
      /usr/local/node_modules doesn't exist or is not a directory
      /usr/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /usr/local/app/node_modules
        using description file: /usr/local/app/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /usr/local/app/package.json (relative path: ./node_modules)
          using description file: /usr/local/app/package.json (relative path: ./node_modules/module)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/app/node_modules/module doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/app/node_modules/module.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/app/node_modules/module.js doesn't exist
            as directory
              /usr/local/app/node_modules/module doesn't exist

```
And continues something like 1000 more lines...  Not sure what I can do about it. Any idea?
Here is my package.json.
```
{
  "name": "app",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve --host=0.0.0.0 --port=80 --disable-host-check --poll=500",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^5.1.0",
    "@angular/cdk": "^5.0.0",
    "@angular/common": "^5.0.0",
    "@angular/compiler": "^5.0.0",
    "@angular/core": "^5.0.0",
    "@angular/forms": "^5.0.0",
    "@angular/http": "^5.0.0",
    "@angular/material": "^5.0.0",
    "@angular/platform-browser": "^5.0.0",
    "@angular/platform-browser-dynamic": "^5.0.0",
    "@angular/router": "^5.0.0",
    "angular-new-router": "github:angular/router",
    "angular2-draggable": "^1.0.7",
    "core-js": "^2.4.1",
    "hammerjs": "^2.0.8",
    "immutable": "^3.8.2",
    "jquery": "^3.2.1",
    "jquery-ui": "^1.12.1",
    "monaco-editor": "^0.10.1",
    "ng-drag-drop": "^4.0.0",
    "rxjs": "^5.5.2",
    "zone.js": "^0.8.14"
  },
  "devDependencies": {
    "@angular/cli": "1.5.2",
    "@angular/compiler-cli": "^5.0.0",
    "@angular/language-service": "^5.0.0",
    "@types/jasmine": "~2.5.53",
    "@types/jasminewd2": "~2.0.2",
    "@types/jquery": "^3.2.16",
    "@types/jqueryui": "^1.11.37",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.2.0",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "ts-node": "~3.2.0",
    "tslint": "~5.7.0",
    "typescript": "~2.4.2"
  }
}
```

I get same error ```Module not found: Error: Can't resolve 'module' in ..node_modules/source-map-support ``` in versions 0.5.16, 0.5.18, and 0.4.18 (versions used in my packages)

--- a/source-map-support.js
+++ b/source-map-support.js
@@ -536,12 +536,8 @@ exports.install = function(options) {
 
   // Support runtime transpilers that include inline source maps
   if (options.hookRequire && !isInBrowser()) {
-    var Module;
-    try {
-      Module = require('module');
-    } catch (err) {
-      // NOP: Loading in catch block to convert webpack error to warning.
-    }
+    // Use dynamicRequire to avoid including in browser bundles
+    var Module = dynamicRequire(module, 'module');
     var $compile = Module.prototype._compile;
 
     if (!$compile.__sourceMapSupport) {

